packages/components/src/mobile/global-styles-context/utils.native.js

```diff
@@ -224,7 +224,12 @@ export function parseStylesVariables( styles, mappedValues, customValues ) {
 			const customValuesData = customValues ?? JSON.parse( stylesBase );
 			stylesBase = stylesBase.replace( regex, ( _$1, $2 ) => {
 				const path = $2.split( '--' );
-				if ( has( customValuesData, path ) ) {
+				if (
+					path.reduce(
+						( prev, curr ) => prev && prev[ curr ],
```

Comment:
This will work if a segment is always either an object or a falsy value, but will fail if it's anything else (say, a number). I don't think that's much of a concern here, as it'd be a symptom of a problem elsewhere instead, but I thought I'd point it out.